python/pyspark/sql/readwriter.py

```diff
@@ -545,6 +545,57 @@ def partitionBy(self, *cols):
         self._jwrite = self._jwrite.partitionBy(_to_seq(self._spark._sc, cols))
         return self
 
+    @since(2.2)
+    def bucketBy(self, numBuckets, *cols):
+        """Buckets the output by the given columns on the file system.
```

Comment:
I think just copying it from Scala doc is good enough to prevent overhead of sweeping the documentation when we start to support other operations later.